src/lib/fs/files.c

```diff
@@ -718,6 +718,25 @@ read_file_to_str, (const char *filename, int flags, struct stat *stat_out))
   return string;
 }
 
+/** Attempt to read a file <b>fname</b>. If the file's contents is
+ * equal to the string <b>str</b>, return 0. Otherwise, attempt to
+ * overwrite the file with the contents of <b>str</b> and return
+ * the value of write_str_to_file().
+ */
+int
+write_str_if_not_equal(const char *fname, const char *str)
```

Comment:
This one might be clearer for future readers as write_str_to_file_if_not_equal. That way it's more obvious that it's like write_str_to_file, just a special case of it.